builtin/logical/database/rotation.go

```diff
@@ -64,6 +64,29 @@ func (b *databaseBackend) populateQueue(ctx context.Context, s logical.Storage)
 			continue
 		}
 
+		// If an account's NextVaultRotation period is nil, it means that the
+		// role was created before we added the `NextVaultRotation` field. In this
+		// case, we need to calculate the next rotation time based on the
+		// LastVaultRotation and the RotationPeriod. However, if the role was
+		// created with skip_import_rotation set, we need to use the current time
+		// instead of LastVaultRotation because LastVaultRotation is 0
+		if role.StaticAccount.NextVaultRotation.IsZero() {
+			log.Debug("NextVaultRotation is zero", roleName)
+			// Previously skipped import rotation roles had a LastVaultRotation value of zero
+			if role.StaticAccount.LastVaultRotation.IsZero() {
+				role.StaticAccount.SetNextVaultRotation(time.Now())
```

Comment:
Does this mean that the Vault will rotated the credential right away? If so, this feels like it my be a surprising side-effect of upgrading to "this" version of Vault.

Reply:
[That function](https://github.com/hashicorp/vault/blob/main/builtin/logical/database/path_roles.go#L998) is a little confusing. What it's actually doing is adding `time.Now` to the rotation period. So it's not setting it to now.

Reply:
Thanks for the explanation. I agree, the naming of that function is a bit confusing. We may want to look at renaming it at some point. Something like `SetRotationRelativeTo(...)` or something like that. Not a blocker.

Reply:
I'll make a note of that. Pretty sure the ldap secrets engine has a similar confusing function :)